### PR TITLE
Add support for LLVM and JS

### LINKED ISSUE #10: Use full `wasmer` performance on llvm and js
For llvm, we should use https://crates.io/crates/wasmer-compiler-llvm.

For js environments, (since we compile lurk to wasm32) we should use this https://wasmerio.github.io/wasmer/crates/doc/wasmer/#using-wasmer-in-a-javascript-environment

### PR DESCRIPTION
**Pull request by winston-h-zhang**
Resolves #10.

TODO: Reflect these changes in CI tests

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -26,8 +26,17 @@ pasta_curves = { version = "0.5.1" }
...
 thiserror = "1.0.43"
+
+[target.'cfg(not(target_arch = "wasm32"))'.dependencies]
 wasmer = "4.0.0"
 
+[target.'cfg(all(not(target_arch = "wasm32"), features = "llvm"))'.dependencies]
+wasmer-compiler-llvm = "4.1.1"
+
+[target.'cfg(target_arch = "wasm32")'.dependencies]
+wasmer = { version = "4.0.0", features = ["js"] }
```
Comment (winston-h-zhang) [inline Cargo.toml:37]: I'm not sure any of this is the correct way to do this

**Review by huitseeker — COMMENTED** [review]
We already basically support wasm32 compilation in Lurk, you can certainly take some ideas from there. Here's a skeleton to get you started:
```toml
name: Rust Wasm32 CI

on: [push, pull_request]

jobs:
  test:
    name: Run tests
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2
    - name: Install Rust
      uses: actions-rs/toolchain@v1
      with:
        toolchain: stable
        override: true
        target: wasm32-unknown-unknown
    - name: Check Build
      uses: actions-rs/cargo@v1
      with:
        command: check
        args: --target wasm32-unknown-unknown
    - name: Run Tests
      uses: actions-rs/cargo@v1
      with:
        command: test
        args: --target wasm32-unknown-unknown
```
(we prefer nextest across lurk-lab, the above is just for example)

**Comment by winston-h-zhang** [comment]
For CI, I just left it at wasm building, but otherwise, LGTM

**Review by huitseeker — APPROVED** [review]
We probably want to follow this up shortly with https://github.com/lurk-lab/circom-scotia/pull/13

**Pull request merged**